### ISSUE TITLE
Sanitize does not work with custom error

### ISSUE THREAD

**Issue by kaankoken** [greyblake/nutype#181]
Hello, when I try to use custom error handling, I observed that `sanitize` does not work at all. Is that intentional? 

```rust 
// Does not work
use thiserror::Error;

#[derive(Error, Debug)]
pub enum FieldErrors {
	#[error("Config field `{0}` is not presented")]
	ConfigField(String),

	#[error("Config fields `{0}` are not presented")]
	ConfigFields(String),

	#[error("Empty field on {0} is not allowed")]
	EmptyField(String),
}

#[nutype(sanitize(trim), validate(with = validate_owner, error = FieldErrors), derive(Debug, PartialEq))]
struct Owner(String);

fn validate_owner(value: &str) -> Result<(), FieldErrors> {
	if value.is_empty() {
		return Err(FieldErrors::EmptyField(String::from("owner")));
	}

	Ok(())
}
```

```rust 
// Working example
use thiserror::Error;

#[derive(Error, Debug)]
pub enum FieldErrors {
	#[error("Config field `{0}` is not presented")]
	ConfigField(String),

	#[error("Config fields `{0}` are not presented")]
	ConfigFields(String),

	#[error("Empty field on {0} is not allowed")]
	EmptyField(String),
}

#[nutype(validate(with = validate_owner, error = FieldErrors), derive(Debug, PartialEq))]
struct Owner(String);

fn validate_owner(value: &str) -> Result<(), FieldErrors> {
	if value.trim().is_empty() {
		return Err(FieldErrors::EmptyField(String::from("owner")));
	}

	Ok(())
}
```

**Comment by greyblake** [comment]
@kaankoken Hi, thanks for reporting, though I could not reproduce what's not working.

I took your example, and the sanitization is taking a place: the spaces are trimmed before validation is applied.

```rs
use thiserror::Error;
use nutype::nutype;

#[derive(Error, Debug, PartialEq)]
pub enum FieldErrors {
	#[error("Config field `{0}` is not presented")]
	ConfigField(String),

	#[error("Config fields `{0}` are not presented")]
	ConfigFields(String),

	#[error("Empty field on {0} is not allowed")]
	EmptyField(String),
}

#[nutype(
    sanitize(trim),
    validate(with = validate_owner, error = FieldErrors),
    derive(Debug, PartialEq)
)]
struct Owner(String);

fn validate_owner(value: &str) -> Result<(), FieldErrors> {
	if value.is_empty() {
		return Err(FieldErrors::EmptyField(String::from("owner")));
	}

	Ok(())
}

fn main() {
    // Success: the string is trimmed
    let john = Owner::try_new("  John  ").unwrap();
    assert_eq!(john.into_inner(), "John");

    // Error: the string with empty spaces is empty -> Error
    let err = Owner::try_new("  ").unwrap_err();
    assert_eq!(err, FieldErrors::EmptyField(String::from("owner")));
}
```

**Comment by kaankoken** [comment]
I was using nightly 1.83.0, let me try again in stable version. I'll come back to you.

**Comment by greyblake** [comment]
Works for me the same way (correctly) with `rustc 1.83.0-nightly (12b26c13f 2024-09-07)`

**Comment by kaankoken** [comment]
I think it is my bad. I was also testing the `validator` methods separately apart from `Owner::try_from()`. I am very sorry. Thanks for your kind response. @greyblake
